### PR TITLE
vdirsyncer: add urlCommand and userNameCommand options

### LINKED ISSUE #4399: vdirsyncer: secret url
### Description

According to the [manual](https://vdirsyncer.pimutils.org/en/stable/keyring.html) any kind of parameter in a storage section can be combined with `.command`. I'd like to combine it with `url` option to not have plaintext url in my public config repo nor in nix store. I'd like to use it with outlook, but as I know it's a common practice for calendar providers to have a possibility to generate a secret url to access a personal calendar. This will require to add `urlCommand` into [accounts](https://github.com/nix-community/home-manager/blob/f5c15668f9842dd4d5430787d6aa8a28a07f7c10/modules/accounts/calendar.nix#L48). As I see there is a [commented userNameCommand](https://github.com/nix-community/home-manager/blob/f5c15668f9842dd4d5430787d6aa8a28a07f7c10/modules/accounts/calendar.nix#L60) which was disabled in the [original PR](https://github.com/nix-community/home-manager/pull/4078). What was the reason to disable it? cc @teto

### PR DESCRIPTION
**Pull request by kurnevsky**
### Description

Fixes #4399 

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [ ] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

#### Maintainer CC

@pjones @teto

### CODE REVIEW

File: modules/accounts/calendar.nix [hunk 1]
```diff
@@ -51,20 +51,27 @@ let
...
 
-      # userNameCommand = mkOption {
-      #   type = types.nullOr (types.listOf types.str);
-      #   default = null;
-      #   example = [ "~/get-username.sh" ];
-      #   description = ''
-      #     A command that prints the user name to standard output.
-      #   '';
-      # };
+      userNameCommand = mkOption {
```
Comment (rycee) [inline modules/accounts/calendar.nix:67]: I guess `userName` and `userNameCommand` are mutually exclusive. I.e., we should have an assertion that causes an error if the user has given both. Same holds for `url` and `urlCommand`.

File: modules/programs/vdirsyncer.nix [hunk 2]
```diff
@@ -24,8 +24,7 @@ let
 
   localStorage = a:
     filterAttrs (_: v: v != null)
-    ((getAttrs [ "type" "fileExt" "encoding" ] a.local) // {
-      path = a.local.path;
+    ((getAttrs [ "type" "fileExt" "encoding" "path" ] a.local) // {
```
Comment (rycee) [inline modules/programs/vdirsyncer.nix:27]: This seems unrelated to the urlCommand and userNameCommand support? If so then I would suggest putting this change in a separate PR.

**Comment by rycee** [comment]
I think this change is fair enough but I'm concerned that urlCommand and userNameCommand are pretty specific for vdirsyncer. That is, if no other software support reading URL and username from commands then perhaps it would be better to put these options as vdirsyncer specific options?

File: modules/programs/vdirsyncer.nix (hunk 2 above)
Comment (kurnevsky) [reply modules/programs/vdirsyncer.nix:27]: It is, just noticed that it was unnecessary to set it explicitly. I wouldn't bother to create a separate PR for this, so I guess it will remain as is forever :)

File: modules/accounts/calendar.nix (hunk 1 above)
Comment (kurnevsky) [reply modules/accounts/calendar.nix]: Added.

**Comment by kurnevsky** [comment]
Actually what other software that supports remote accounts? If I don't miss something it's only vdirsyncer :)

**Comment by kurnevsky** [comment]
Still relevant.

**Comment by kurnevsky** [comment]
Why is it stale?

**Comment by rycee** [comment]
Sorry about the delay!
